conf/base/bblayers.conf

```diff
@@ -3,22 +3,24 @@
 POKY_BBLAYERS_CONF_VERSION = "2"
 
 BBPATH = "${TOPDIR}"
+BSPDIR := "${@os.path.abspath(os.path.dirname(d.getVar('FILE', True)) + '/../..')}"
+
 BBFILES ?= ""
 
 BBLAYERS ?= " \
-  ##OEROOT##/meta \
-  ##OEROOT##/meta-poky \
-  ##OEROOT##/meta-yocto-bsp \
-  ##OEROOT##/meta-qt4 \
-  ##OEROOT##/meta-qt5 \
-  ##OEROOT##/meta-uav \
-  ##OEROOT##/meta-ros \
-  ##OEROOT##/meta-intel-aero \
-  ##OEROOT##/meta-intel-aero-connectivity \
-  ##OEROOT##/meta-intel-realsense \
-  ##OEROOT##/intel-aero-samples \
-  ##OEROOT##/meta-openembedded/meta-oe \
-  ##OEROOT##/meta-openembedded/meta-python \
-  ##OEROOT##/meta-openembedded/meta-networking \
-  "
+  ${BSPDIR}/sources/poky/meta \
```

Comment:
The only problem with putting all of the sources inside the poky folder is that, since the new layer directories and the build directories are not part of a poky gitignore it pollutes the poky repository.
Having a sources dir allows to keep each layer clean (including poky) and that comes with lots of benefits. Easier searching, being able to check out other branches, etc.
I should also say that the sources folder isn't an original idea of mine. This is the structure that the freescale community BSP project uses, we could point to their documentation as reference too. They are a fantastic learning resource. Here's a few links:
http://freescale.github.io/doc/release-notes/2.0/
https://github.com/Freescale/fsl-community-bsp-platform